### PR TITLE
[WIP] Resolves syft.lib.python.Any in the code base

### LINKED ISSUE #5369: Resolve `syft.lib.python.Any` in the code base
## Description
Go through and find all uses of `syft.lib.python.Any` and either fix them if the functionality already exists, or add a TODO and Issue to github to describe them.

Some of these can be solved with things like a `Numeric` `UnionGenerator`.

## Definition of Done
All our `syft.lib.python.Any` remaining in the code base have been checked and fixed or noted down as TODO.

### PR DESCRIPTION
**Pull request by ArtistBanda**
## Description
fixes #5369 
Fixing all the uses of `syft.lib.python.Any` in the codebase.
(Only changed one occurrence for review, will update all if accepted) 

## Affected Dependencies
None

## How has this been tested?
Checked using pre-written tests for python libs and ast.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by tudorcebere** [comment]
Hi @ArtistBanda!

For all the changes that are made, please add a test on them and double-check that wrapping iterators is not pointless (double wrapping might occur). Down below in the AST I am wrapping the iterator so it might not be useful. Keep going, you are on the right track. :smile:

**Comment by ArtistBanda** [comment]
> Hi @ArtistBanda!
> 
> For all the changes that are made, please add a test on them and double-check that wrapping iterators is not pointless (double wrapping might occur). Down below in the AST I am wrapping the iterator so it might not be useful. Keep going, you are on the right track.

Well for resolving `syft.lib.python.Any` in the `create_python_ast` for methods like `__getitem__()` and `pop()`, what should it be replaced with?
Should `syft.lib.python.Any` be replaced with 
```
UnionGenerator[
    "syft.lib.python.Int",
    "syft.lib.python.Float",
    "syft.lib.python.String",
    "torch.nn.Parameter",
    "torch.Tensor",
],
```

**Comment by tudorcebere** [comment]
Yeah, maybe it would be worth making an Alias like: Generic Syft Type or something like that?

**Comment by ArtistBanda** [comment]
@tudorcebere I think this resolves the issue. About the test cases could you guide me a bit more?

**Comment by ArtistBanda** [comment]
The PR is complete from my side @tudorcebere @madhavajay . If there is anything more required please let me know.

**Comment by tudorcebere** [comment]
Closing this due to inactivity. Please open a new PR with the new monorepo structure/updates. Thanks for the work, we will design a new implementation in the backlog.
